js/components/cross-section.tsx

```diff
@@ -1,32 +1,67 @@
 import React from "react";
 import { inject, observer } from "mobx-react";
 import { TransitionGroup, CSSTransition } from "react-transition-group";
-import SmallButton from "./small-button";
-import CrossSection3D from "./cross-section-3d";
+import CrossSection3D, { IOnCreateSceneParams } from "./cross-section-3d";
 import { BaseComponent, IBaseProps } from "./base";
+import { CAMERA_ZOOM_STEP, MAX_CAMERA_ZOOM, MIN_CAMERA_ZOOM } from "../types";
+import ModelCloseIconSVG from "../../images/model-close-icon.svg";
+import ModelResetIconSVG from "../../images/model-reset-icon.svg";
+import ZoomInIconSVG from "../../images/zoom-in-icon.svg";
+import ZoomOutIconSVG from "../../images/zoom-out-icon.svg";
 
 import "../../css/cross-section.less";
 
 export const CROSS_SECTION_TRANSITION_LENGTH = 400; // ms
 
-interface IState {}
+interface IState {
+  cameraZoom: number;
+}
 
 @inject("simulationStore")
 @observer
 export default class CrossSection extends BaseComponent<IBaseProps, IState> {
+  camera: THREE.OrthographicCamera;
+
+  handleCreateScene = (params: IOnCreateSceneParams) => {
+    this.camera = params.camera;
+  };
+
+  handleZoomIn = () => {
+    if (this.camera) {
+      this.camera.zoom = Math.min(MAX_CAMERA_ZOOM, this.camera.zoom + CAMERA_ZOOM_STEP);
+      this.camera.updateProjectionMatrix();
+      this.setState({ cameraZoom: this.camera.zoom });
+    }
+  };
+
+  handleZoomOut = () => {
+    if (this.camera) {
+      this.camera.zoom = Math.max(MIN_CAMERA_ZOOM, this.camera.zoom - CAMERA_ZOOM_STEP);
+      this.camera.updateProjectionMatrix();
+      this.setState({ cameraZoom: this.camera.zoom });
+    }
+  };
+
+  handleResetCamera = () => {
+    this.camera.zoom = 1;
```

Comment:
It's a minor issue, but it's a bit harsh compared to the smooth camera angle animation. 😉

Reply:
True, but I wasn't sure it was worth the effort to implement the animation. It didn't seem to look _too_ bad when I was testing it.